crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,357 @@
+
+/// An [`AccountInterface`] describes the exported, callable procedures of an account.
+///
+/// A note script's compatibility with this interface can be inspected to check whether the note may
+/// result in a successful execution against this account.
+pub struct AccountInterface {
+    account_id: AccountId,
+    auth: Vec<AuthScheme>,
+    components: Vec<AccountComponentInterface>,
+}
+
+impl AccountInterface {
+    // CONSTRUCTORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Creates a new [`AccountInterface`] instance from the provided account ID, authentication
+    /// schemes and account code.
+    pub fn new(account_id: AccountId, auth: Vec<AuthScheme>, code: &AccountCode) -> Self {
+        let components = AccountComponentInterface::from_procedures(code.procedures());
+
+        Self { account_id, auth, components }
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns a reference to the account ID.
+    pub fn id(&self) -> &AccountId {
+        &self.account_id
+    }
+
+    /// Returns the type of the reference account.
+    pub fn account_type(&self) -> AccountType {
+        self.account_id.account_type()
+    }
+
+    /// Returns true if the reference account can issue assets.
+    pub fn is_faucet(&self) -> bool {
+        self.account_id.is_faucet()
+    }
+
+    /// Returns true if the reference account is a regular.
+    pub fn is_regular_account(&self) -> bool {
+        self.account_id.is_regular_account()
+    }
+
+    /// Returns true if the reference account is public.
+    pub fn is_public(&self) -> bool {
+        self.account_id.is_public()
+    }
+
+    /// Returns a reference to the vector of used authentication schemes.
+    pub fn auth(&self) -> &Vec<AuthScheme> {
+        &self.auth
+    }
+
+    /// Returns a reference to the set of used component interfaces.
+    pub fn components(&self) -> &Vec<AccountComponentInterface> {
+        &self.components
+    }
+
+    /// Returns [NoteAccountCompatibility::Maybe] if the provided note is compatible with the
+    /// current [AccountInterface], and [NoteAccountCompatibility::No] otherwise.
+    pub fn is_compatible_with(&self, note: &Note) -> NoteAccountCompatibility {
+        if let Some(well_known_note) = WellKnownNote::from_note(note) {
+            if well_known_note.is_compatible_with(self) {
+                NoteAccountCompatibility::Maybe
+            } else {
+                NoteAccountCompatibility::No
+            }
+        } else {
+            verify_note_script_compatibility(note.script(), self.component_procedure_digests())
+        }
+    }
+
+    /// Returns a digests vector of all procedures from all account component interfaces.
+    pub(crate) fn component_procedure_digests(&self) -> BTreeSet<Digest> {
```

Comment:
The comment here needs to be updated (we are not returning a vector. Also, I'd rename this into `get_procedure_digests()`.